### ISSUE TITLE
Bogus `cannot be captured` error

### ISSUE THREAD

**Issue by dawkot** [nim-lang/Nim#15267]
I don't think this is an intented limitation.

### Example
```nim
proc foo(x: var int, y = x) = discard

proc bar(x: var int) =
  foo x

var global: int

bar global
```

### Current Output
```
Error: 'x' is of type <var int> which cannot be captured as it would violate memory safety,
using '-d:nimWorkaround14447' helps in some cases
```

### Additional Information
* `-d:nimWorkaround14447'` doesn't help
* ff13f8cc3c74d4048b6c2604fe4b8e5365f5c13f

**Comment by narimiran** [comment]
Using the latest Nim devel version, I cannot reproduce this error.

@dawkot what is your Nim version? If you can update to the latest devel, please see if it is also (not) there for you.

**Comment by Clyybber** [comment]
Fixed in #15270
